test/CMakeLists.txt

```diff
@@ -66,6 +66,19 @@ set_target_properties(catch_main PROPERTIES
 )
 target_include_directories(catch_main PRIVATE "thirdparty/catch")
 
+# https://stackoverflow.com/questions/2368811/how-to-set-warning-level-in-cmake
+if(MSVC)
+    # Force to always compile with W4
```

Comment:
Why do you want to force the `/W4` flag?

Reply:
With /W4 always enabled we can ensure no warnings will be emitted when adding this library to an existing project. It feels like a library such as this one should support /W4 with no issues. There are other warnings in unit tests even with /W3, but that would be for another PR. At least there is visibility now.

Reply:
Ok, thanks. I realized we're up to some 50 warnings now...

Reply:
I would really appreciate any help with MSVC regarding this warnings, as I have no experience here.